matchms/filtering/require_precursor_below_mz.py

```diff
@@ -0,0 +1,26 @@
+from ..typing import SpectrumType
+
+
+def require_precursor_below_mz(spectrum: SpectrumType, max_mz: float = 1000) -> SpectrumType:
+
+    """Returns None if the precursor_mz of a spectrum is above
+       max_mz.
+
+    Args:
+    -----
+    spectrum:
+        Input spectrum.
+    max_mz:
+        Maximum mz value for the precursor mz of a spectrum.
+        All precursor mz values greater or equal to this
+        will return none. Default is 1000.
+    """
+
+    assert max_mz >= 0, "max_mz must be a positive floating point."
+    precursor_mz = spectrum.get("precursor_mz")
+    if precursor_mz:
+
+        if precursor_mz >= max_mz:
+            return None
```

Comment:
```suggestion
    if precursor_mz and precursor_mz >= max_mz:
        return None
```

Reply:
SonarCloud complains about a "code smell" which is the tools' way of suggesting code improvements. I think the suggested change would solve that.

Reply:
My reason for the nested if statements was in case the precursor_mz was missing (so would be None, rather than 0).  But I suppose we can ignore that?

Reply:
That will be dealt with as by Python. 
If you precursor_mz is None than it will not go further to the ``and ...`` part.
(so if you change the order: ``if precursor_mz >= max_mz and precursor_mz`` that would indeed fail for ``None``)

Reply:
Ah I see, so it wouldn't give a TypeError.